### PR TITLE
Date(time) pk

### PR DESCRIPTION
**Pull request by jkavalik**

### CODE REVIEW

**Comment by hrach** [comment]
Please, as you can see, since updates to other rdbms' it is best to keep changes as small as possible. Please, fix all build issues. Thx.

**Comment by jkavalik** [comment]
Yes, working on it.
OTOH the tests failure seems unrelated to me 
```
Service container mysql80 failed.
Service container mssql failed.
```
for MySQL specifically
```
Error:  2024-10-08T09:40:01.976192Z 0 [ERROR] [MY-000067] [Server] unknown variable 'default-authentication-plugin=mysql_native_password'.
```
I will look into that too

**Comment by hrach** [comment]
Oh, sorry, that something that broke and I hadn't time to fix it. If you don't want, let fixing this on me.

File: tests/cases/integration/Entity/entity.compositePK.phpt [hunk 1]
```diff
@@ -77,6 +77,7 @@ class EntityCompositePKTest extends DataTestCase
 		$this->orm->clear();
 
 		$res = $this->orm->userStatsX->getBy(['date' => new DateTime('2019-01-01')]);
+		Assert::notNull($res);
```
Comment (hrach) [inline tests/cases/integration/Entity/entity.compositePK.phpt:80]: getByChecked?

**Comment by hrach** [comment]
Awesome <3

**Comment by jkavalik** [comment]
I replaced getBy(Id)/assertNotNull by their checked variant in the files tests I touched (except the case where it would lead to unused variable).

**Comment by hrach** [comment]
@jkavalik could you please fixup relevant commits, i.e. keep split fixing CI; and then your fix with tests in another commit? thx

**Comment by jkavalik** [comment]
Fixed a bad conflict resolution..

**Comment by hrach** [comment]
Thank you a lot!

**Pull request merged**